src/cardanoLauncher.ts

```diff
@@ -230,18 +230,22 @@ export class Launcher {
    * server is ready to accept requests.
    */
   start(): Promise<Api> {
-    this.nodeService.start();
-    this.walletService.start();
-
     const stopWaiting = () =>
       this.nodeService.getStatus() > ServiceStatus.Started ||
       this.walletService.getStatus() > ServiceStatus.Started;
 
-    this.waitForApi(stopWaiting).then(() => {
-      this.walletBackend.events.emit('ready', this.walletBackend.getApi());
-    });
-
```

Comment:
what about simply `return this.waitForApi(stopWaiting).then(() => { ...`, so that, the "catch" is implicitly passed down to the caller of `start()`.

Reply:
We need a way to cancel `waitForApi` in the event of wallet or node exiting.

When `waitForApi` has been cancelled, it rejects the promise. There is no reason to act on that. It's already been handled elsewhere.

I can see that this is confusing. I will change `waitForApi` to use a simple callback rather than a promise.